### PR TITLE
Fix sidenav active state not showing and icon placement

### PR DESCRIPTION
**Pull request by stanislavgeorgiev**
Closes carbon-design-system/carbon-components-angular#1889

Before

image

After

image

Before

image

After

image

### CODE REVIEW

File: src/ui-shell/sidenav/sidenav-menu.component.ts [hunk 1]
```diff
@@ -113,10 +113,6 @@ export class SideNavMenu implements AfterContentInit, OnDestroy {
 	}
 
 	protected findActiveChildren() {
-		if (this.sidenavItems.some(item => item.active)) {
-			this.hasActiveChild = true;
-		} else {
-			this.hasActiveChild = false;
-		}
```
Comment (Akshat55) [inline src/ui-shell/sidenav/sidenav-menu.component.ts:120]: Nice catch!

File: src/ui-shell/sidenav/sidenav-item.component.ts [hunk 2]
```diff
@@ -23,6 +23,9 @@ import { Router } from "@angular/router";
 			<a
 				class="bx--side-nav__link"
 				[href]="href"
+				[ngClass]="{
+					'.bx--side-nav__link--current': active
```
Comment (Akshat55) [inline src/ui-shell/sidenav/sidenav-item.component.ts:27]:
```suggestion
					'bx--side-nav__link--current': active
```

**Review by Akshat55 — CHANGES_REQUESTED** [review]

File: src/ui-shell/sidenav/sidenav-menu.component.ts (hunk 1 above)
Comment (stanislavgeorgiev) [reply src/ui-shell/sidenav/sidenav-menu.component.ts:120]: Thanks :)
I was looking for how things work when I stumbled upon this block :D

File: src/ui-shell/sidenav/sidenav-item.component.ts (hunk 2 above)
Comment (stanislavgeorgiev) [reply src/ui-shell/sidenav/sidenav-item.component.ts:27]: Aaaaa.... OK
That didn't happen

**Review by Akshat55 — APPROVED** [review]

**Review by zvonimirfras — APPROVED** [review]

**Pull request merged**